### PR TITLE
add Conv2D/Conv2DTranspose/Conv3D/Conv3DTranspose in paddle.nn.layer

### PR DESCRIPTION
**Pull request by iclementine**
1. add add Conv2D/Conv2DTranspose/Conv3D/Conv3DTranspose in paddle.nn.layer. 
2. all the 4 layers support `data_format` and all kinds of `padding`s.
3. `Conv2DTranspose` and `Conv3DTranspose` have keyword argument `output_size`.
4. bugfix:  `dygraph.Conv3DTranspose` has no bias by default.
5. add unittests for all the 4 layers added.

### CODE REVIEW

**Review by kuke — APPROVED** [review]
LGTM

Seems that there are a lot problems in the original docstrings, and better to new another pull request to refine them

**Comment by iclementine** [comment]
> LGTM
> 
> Seems that there are a lot problems in the original docstrings, and better to new another pull request to refine them

OK, I will refine the docstrings in a new PR.

**Review by raindrops2sea — APPROVED** [review]
LGTM

**Comment by shippingwang** [comment]
This is an apples-to-apples comparison between torch and paddle:
and run test code on ```Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz``` and ```NVIDIA P40``` single card
please install paddlepaddle version later than this PR and torch at first.

```python
def main():
    import paddle
    from paddle import fluid
    import paddle.fluid.dygraph as dg
    import paddle.nn as nn
    import numpy as np
    import time
    import torch.nn as torchnn
    import torch

    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
      with dg.guard():
        x_var = dg.to_variable(input)
        conv = nn.Conv2D(4,6,(3,3))
        y_var = conv(x_var)
    end_time = time.time()
    paddle_cpu_cost = (end_time - start_time) / 10000
    print('paddle cpu time: ', paddle_cpu_cost*1000, 'ms')

    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
      with dg.guard(fluid.CUDAPlace(0)):
        x_var = dg.to_variable(input)
        conv = nn.Conv2D(4,6,(3,3),use_cudnn=True)
        y_var = conv(x_var)
    end_time = time.time()
    paddle_gpu_cost = (end_time - start_time) / 10000
    print('paddle gpu time(use_cudnn=True): ', paddle_gpu_cost*1000, 'ms')

    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
     with dg.guard(fluid.CUDAPlace(0)):
        x_var = dg.to_variable(input)
        conv = nn.Conv2D(4,6,(3,3),use_cudnn=False)
        y_var = conv(x_var)
    end_time = time.time()
    paddle_gpu_cost = (end_time - start_time) / 10000
    print('paddle gpu time(use_cudnn=False): ', paddle_gpu_cost*1000, 'ms')


    ######################################################################

    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
      conv = torchnn.Conv2d(4,6,(3,3))
      input_var = torch.from_numpy(input)
      output = conv(input_var)
    end_time = time.time()
    pytorch_cpu_cost = (end_time - start_time) / 10000
    print('pytorch cpu time: ', pytorch_cpu_cost*1000, 'ms')

    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
      conv = torchnn.Conv2d(4,6,(3,3)).cuda()
      input_var = torch.from_numpy(input).cuda()
      output = conv(input_var)
    end_time = time.time()
    pytorch_gpu_cost = (end_time - start_time) / 10000
    print('pytorch gpu time: ', pytorch_gpu_cost*1000, 'ms')

    torch.backends.cudnn.benchmark=True
    start_time = time.time()
    for _ in range(10000):
      input = np.random.uniform(-1, 1, (2,4,8,8)).astype('float32')
      conv = torchnn.Conv2d(4,6,(3,3)).cuda()
      input_var = torch.from_numpy(input).cuda()
      output = conv(input_var)
    end_time = time.time()
    pytorch_gpu_cost = (end_time - start_time) / 10000
    print('pytorch gpu time: (backends.cudnn.benchmark=True)', pytorch_gpu_cost*1000, 'ms')
main()
```
result: 
image
